README.md

```diff
@@ -136,6 +137,36 @@ julia> central_fdm(5, 1, max_range=9e-1)(ε -> logdet(x .+ ε .* v), 0) - sum(1
 -1.283417816466681e-13
 ```
 
+### Dealing with Numerical Noise
+
+It could be the case that the function `f` you'd like compute the derivative of
+suffers from numerical noise.
+For example, `f(x)` could be computed through some iterative procedure with some
+error tolerance `ε`.
+In such cases, finite difference estimates can fail catastrophically.
+To illustrate this, consider `sin_noisy(x) = sin(x) * (1 + 1e-6 * randn())`.
+Then
+
+```julia
+julia> central_fdm(5, 1)(sin_noisy, 1) - cos(1)
+-0.027016678790599657
+```
+
+which is a terrible performance.
+To deal with this, you can set the keyword argument `factor`, which specifies
+numerical noise on the function evaluation relative to the machine epsilon.
+In this example, the noise lies in `[-2, 2]` (roughly) and the machine epsilon
+is `eps(Float64) ≈ 2.22e-16`, so `factor = 2e-6 / 2e-16 = 1e10` should be
```

Comment:
sorry i didn't quite follow: where does the `2e-6` term come from?

Reply:
I've rewritten the sentence. Is this more clear now?

Reply:
Yep, thanks!